ethereum/utils.py

```diff
@@ -35,6 +35,9 @@ def to_string_for_regexp(value):
         return str(value)
     unicode = unicode
 
+    def bytearray_to_bytestr(value):
+        return bytes(''.join(chr(c) for c in value))
```

Comment:
I don't think this is necessary. A simple `str(value)` *should* be enough. 
```
str(bytearray([0, 1, 2, 254, 255]))
'\x00\x01\x02\xfe\xff'
```
Seems  to work fine in Python 2.

And since `bytes` is an alias to `str` in Python 2 we wouldn't even need the two versions of the function.